core/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -104,7 +105,7 @@ protected void masterOperation(final GetSnapshotsRequest request, ClusterState s
                     }
                 }
                 for (String snapshot : snapshotsToGet) {
-                    SnapshotId snapshotId = new SnapshotId(request.repository(), snapshot);
+                    final SnapshotId snapshotId = snapshotsService.resolveSnapshotId(new SnapshotName(request.repository(), snapshot));
```

Comment:
This means that we query the repository for every name in this loop? Wouldn't it make sense to add a resolveSnapshotIds method that resolves all of them in one go?